ui/sv_icons.py

```diff
@@ -0,0 +1,76 @@
+import bpy
+import os
+import glob
+import bpy.utils.previews
+
+DEBUG = False
+
+
+def logDebug(message, extra=""):
+    if DEBUG:
+        print(message, extra)
+
+# custom icon dictionary
+_custom_icons = {}
+
+_custom_icons_loaded = False
+
+
+def custom_icon(name):
+    logDebug("custom_icon called: ", name)
+    global _custom_icons
+    global _custom_icons_loaded
+
+    if not _custom_icons_loaded:
+        load_custom_icons()
+
+    if name in _custom_icons:
+        return _custom_icons[name].icon_id
+    else:
+        logDebug("No custom icon found for name: ", name)
+        return 0
+
+
+def load_custom_icons():
+    logDebug("load_custom_icons called")
+    global _custom_icons
+    global _custom_icons_loaded
+
+    _custom_icons = bpy.utils.previews.new()
+
+    iconsDir = os.path.join(os.path.dirname(__file__), "icons")
+    iconPattern = "sv_*.png"
+    iconPath = os.path.join(iconsDir, iconPattern)
+    iconFiles = [os.path.basename(x) for x in glob.glob(iconPath)]
+    logDebug(iconFiles)
+
+    for iconFile in iconFiles:
+        iconName = os.path.splitext(iconFile)[0]
+        iconID = iconName.upper()
+        logDebug(iconID)
+        _custom_icons.load(iconID, os.path.join(iconsDir, iconFile), "IMAGE")
+
+    _custom_icons_loaded = True
+
+
+def remove_custom_icons():
+    logDebug("remove_custom_icons called")
+    global _custom_icons
```

Comment:
he said that (iirc ) to prevent the namespace polution when using `import *` , but you shouldn't need to import anything other the `custom_icon(name)` function, `  ( i incorrectly suggested that you import `custom_icons` dict above... but that's not needed since you've made a nice function for that dict to always produce a valid response )

any file that needs a custom icon : 
`from sverchok.ul.sv_icons import custom_icon`